share/spice/today_in_history/today_in_history.js

```diff
@@ -1,33 +1,141 @@
-function ddg_spice_today_in_history(api_response) {
+(function (env) {
     "use strict";
+    env.ddg_spice_today_in_history = function(api_response) {
 
-    if (!api_response) {
-        return Spice.failed('today_in_history');
-    }
+        if (!api_response) {
+            return Spice.failed('today_in_history');
+        }
 
-    var item = $( $.parseXML(api_response) ).find('item');
+        // Extract our query
+        var script = $('[src*="/js/spice/today_in_history/"]')[0],
+            source = $(script).attr('src'),
+            ourquery = source.match(/today_in_history\/([^\/]+)/)[1];
+            ourquery = ourquery.replace("_", " ");
 
-    if (!item) {
-        return Spice.failed('today_in_history');
-    }
+        // Extract data
+        var item;
+        for (var key in api_response.query.pages) {
```

Comment:
Check `api_response.query` before accessing the `pages` property? It might be unlikely that the API would change, but if it does, this could break things.